### PR TITLE
Generic IAM commands documentation

### PR DESCRIPTION
**Pull request by Arsenikr**
## Status
In Progress

## Description
Documentation of the generic IAM commands

### CODE REVIEW

File: docs/integrations/generic-commands-iam.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+title: Generic IAM Commands
+---
+
+
+## Background and motivation
+
+IAM integrations generally do not support CRUD (Create, Retrieve, Update, and Disable/Delete) operations for general user accounts or provisioning users. These integrations should adhere to a standard as the workflows can be generalized or completely customized to fit into an organization's environment.
+
+## Mappers
+In order to properly map between the generic IAM user fields and the data from the integration, we need to define an incoming [Mapper](../incidents/incident-classification-mapping#map-event-attributes-to-fields)
```
Comment (dantavori) [inline docs/integrations/generic-commands-iam.md:12]: incoming for get-user, outgoing for update/create.
I think this is also the place to explain that their names should be the default values for "mapper-in" and "mapper-out" parameters.

File: docs/integrations/generic-commands-iam.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+**Input:** User profile which contains the email of the users we would like to retrieve  
+**Input Examples:**
+```
+!iam-get-user user-profile=`{"email": "demistotest@paloaltonetworks.com"}`
+```
+
+**Sample output:**
+```json
+{
+  "IAM.Vendor.": {
```
Comment (dantavori) [inline docs/integrations/generic-commands-iam.md:30]:
```suggestion
  "IAM.Vendor": {
```

File: docs/integrations/generic-commands-iam.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    "instanceName": "Instance name here",
+    "success": true,
+    "active": true,
+    "id": "Zoom Unique Id here",
+    "username": "demistotest@paloaltonetworks.com",
+    "email": "demistotest@paloaltonetworks.com",
+    "errorCode": "null for success response",
+    "errorMessage": "null for success response",
+    "details": {
+      "Zoom Payload can go as is here in json format": ""
```
Comment (dantavori) [inline docs/integrations/generic-commands-iam.md:41]:
```suggestion
    "id": "Unique Id here",
    "username": "demistotest@paloaltonetworks.com",
    "email": "demistotest@paloaltonetworks.com",
    "errorCode": "null for success response",
    "errorMessage": "null for success response",
    "details": {
      "Payload from vendor can go as is here in json format": ""
```

**Review by dantavori — CHANGES_REQUESTED** [review]
looks good 👍🏽

File: docs/integrations/generic-commands-iam.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
+---
+id: generic-commands-iam
+title: Generic IAM Commands
+---
+
+
+## Background and motivation
+
+IAM integrations generally do not support CRUD (Create, Retrieve, Update, and Disable/Delete) operations for general user accounts or provisioning users. These integrations should adhere to a standard as the workflows can be generalized or completely customized to fit into an organization's environment.
```
Comment (anara123) [inline docs/integrations/generic-commands-iam.md:9]:
```suggestion
IAM integrations generally support CRUD (Create, Retrieve, Update, and Disable/Delete) operations for general user accounts or provisioning users. These integrations should adhere to a standard as the workflows can be generalized or completely customized to fit into an organization's environment.
```

File: docs/integrations/generic-commands-iam.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+## Mappers
+In order to properly map between the generic IAM user fields and the data from the integration, we need to define an incoming [Mapper](../incidents/incident-classification-mapping#map-event-attributes-to-fields) for the **get-user** command and an outgoing mapper for the create, update and disable commands. 
+
+The mapper names should be the default values for "mapper-in" and "mapper-out" parameters of the inegration instance.
```
Comment (anara123) [inline docs/integrations/generic-commands-iam.md:14]:
```suggestion
The mapper names should be the default values for "mapper-in" and "mapper-out" parameters of the integration instance. 
```

File: docs/integrations/generic-commands-iam.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+## Background and motivation
+
+IAM integrations generally do not support CRUD (Create, Retrieve, Update, and Disable/Delete) operations for general user accounts or provisioning users. These integrations should adhere to a standard as the workflows can be generalized or completely customized to fit into an organization's environment.
+
+## Mappers
+In order to properly map between the generic IAM user fields and the data from the integration, we need to define an incoming [Mapper](../incidents/incident-classification-mapping#map-event-attributes-to-fields) for the **get-user** command and an outgoing mapper for the create, update and disable commands. 
+
+The mapper names should be the default values for "mapper-in" and "mapper-out" parameters of the inegration instance. 
+
+For integrations that support the [SCIM](http://www.simplecloud.info/) protcol, you can use the built in SCIM mappers in the IAM-SCIM pack.
```
Comment (anara123) [inline docs/integrations/generic-commands-iam.md:16]:
```suggestion
For integrations that support the [SCIM](http://www.simplecloud.info/) protocol, you can use the built in SCIM mappers in the IAM-SCIM pack.  
```

File: docs/integrations/generic-commands-iam.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    "errorCode": "null for success response",
+    "errorMessage": "null for success response",
+    "details": {
+      "Payload can go as is here in json format": ""
+    }
+  }
+}
+```
+
+## **iam-create-user**
```
Comment (anara123) [inline docs/integrations/generic-commands-iam.md:51]:
```suggestion
### **iam-create-user**
```

File: docs/integrations/generic-commands-iam.md [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+    "errorCode": "null for success response",
+    "errorMessage": "null for success response",
+    "details": {
+      "Payload can go as is here in json format": ""
+    }
+  }
+}
+```
+
+## **update-user**
```
Comment (anara123) [inline docs/integrations/generic-commands-iam.md:83]:
```suggestion
### **update-user**
```

File: docs/integrations/generic-commands-iam.md [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+    "errorCode": "null for success response",
+    "errorMessage": "null for success response",
+    "details": {
+      "Payload can go as is here in json format": ""
+    }
+  }
+}
+```
+
+## **iam-disable-user**
```
Comment (anara123) [inline docs/integrations/generic-commands-iam.md:117]:
```suggestion
### **iam-disable-user**
```

File: docs/integrations/generic-commands-iam.md [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+  }
+}
+
+```
+
+## Integrations for reference
+
+[HelloIAMWorld](https://github.com/demisto/content/tree/master/Packs/HelloIAMWorld/Integrations/HelloIAMWorld) 
+[Okta](https://github.com/demisto/content/tree/master/Packs/Okta/Integrations/Okta_IAM)
+[ServiceNow](https://github.com/demisto/content/tree/master/Packs/ServiceNow/Integrations/ServiceNow_IAM)
```
Comment (anara123) [inline docs/integrations/generic-commands-iam.md:156]:
```suggestion
[HelloIAMWorld](https://github.com/demisto/content/tree/master/Packs/HelloIAMWorld/Integrations/HelloIAMWorld)  
[Okta](https://github.com/demisto/content/tree/master/Packs/Okta/Integrations/Okta_IAM)  
[ServiceNow](https://github.com/demisto/content/tree/master/Packs/ServiceNow/Integrations/ServiceNow_IAM) 
 ```

File: docs/integrations/generic-commands-iam.md [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+    "details": {
+      "Payload can go as is here in json format": ""
+    }
+  }
+}
+```
+
+## **iam-create-user**
+
+**Description:** creates a user
```
Comment (anara123) [inline docs/integrations/generic-commands-iam.md:53]: Creates a user, and if user exits then updates the user. If user exits but disabled, then enable the user.

File: docs/integrations/generic-commands-iam.md [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+**Input:** User Profile with the new user data
+
+**Input Examples:**
+
+```
+!iam-create-user user-profile=`{"email": "demistotest@paloaltonetworks.com", "givenname": "test", "surname": "test"}`
+```
+
+**Sample output:**
+```json
```
Comment (anara123) [inline docs/integrations/generic-commands-iam.md:64]: all the commands also return `IAM.UserProfile` object

**Review by anara123 — CHANGES_REQUESTED** [review]
Overall there are a lot of details missing from the doc. New partner/developer should be able to read the doc and be able to develop new IAM integration from scratch

**Comment by glicht** [comment]
@Arsenikr where are we with this?

**Comment by Arsenikr** [comment]
@dantavori took responsibility on it from me because it is a technical docs

**Pull request closed without merging**